### ISSUE TITLE
Add a Connection - SaaS connector's configuration parameters

### ISSUE THREAD

**Issue by adriaaaa** [ethyca/fidesops#761]
### Use Cases

As the owner of a system that contains PII, I need the ability to configure connection details of my application so that Fides can query that system to fulfill a privacy request.

### Assumptions
* Dataset configuration will be done in a separate ticket
* the descriptions that are provided within the fields for a connectors parameters will be addressed in https://github.com/ethyca/fidesops/issues/887
* the color-coded/state-dependent "Active" label will be addressed in a separate story - https://github.com/ethyca/fidesops/issues/886
* ignore the tooltips on each row for now EXCEPT the fides_key row
* use feature flag for this body of work
* backend work that is needed is on this https://github.com/ethyca/fidesops/issues/814
* We already support testing connections on the backend
* We have backend support for sending secrets for each saas connector type - https://github.com/ethyca/fidesops/issues/753
* Ignore the round exclamation icons on the left nav. This work will be covered in https://github.com/ethyca/fidesops/issues/874

### Acceptance Criteria
1. Demonstrate that once a user selects a connector from the list, that can configure the connector's parameters, which include:
* Name (required; must be unique)
* Description (optional)
* that specific connectors configuration parameters 
** this is different for every connector (we have an API that returns the parameters for a connector)
** the API returns which field is optional or required.
2. Show that once a user has inputted all the required fields that they can save 
* The "Test Connection" button should be disabled until the params are saved
3. Confirm that the following are also represented on these pages:
* logo of the connector
* whether the connector is active or disabled.
4. Show that this pages descriptions says "Connect to your Stripe environment by providing credential information below. Once you have saved your connector credentials, you can review what data is included when processing a privacy request in your Dataset configuration."

### Mockups
[Figma](https://www.figma.com/file/MTbZohIGvB4bq2f2wkiDHO/Fidesops-Management-UI?node-id=106%3A1469)

![Image](https://user-images.githubusercontent.com/87208929/179298098-3013c1f7-2dec-4ae7-a89c-c39f61064466.png)

![Image](https://user-images.githubusercontent.com/87208929/179298054-f0c5ea47-65ed-4653-9a02-16b7e29831c7.png)

### Q&A
1. for test requests is it handled on the backend if we can handle the connection. Ans: we support this on the backend
2. verify rate limiting exists. Ans: a problem to solve later.
3. generate unique id based on name. Ans: ticket #814 will verify that the name is unique so we will handle on the save.

**Comment by pattisdr** [comment]
Existing API endpoints

- PATCH `api/v1/connection`: Creates the overall connectionconfig and corresponds to the first few fields on the page.
```json
[
    {"name": "My Stripe Connector",
    "key": "my_stripe_connector",
    "connection_type": "saas",
    "access": "write",
    "description": "This is a connection to stripe"
}]
```
- GET  `api/v1/connection_type/{connection_type}/secret`. Returns the type of secrets you need to supply to connect to a particular connector.  You can use this to dynamically add the inputs below name/description 

For example: GET /api/v1/connection_type/sentry/secret
```json
{
    "title": "sentry_connector_schema",
    "description": "Sentry secrets schema",
    "type": "object",
    "properties": {
        "access_token": {
            "title": "Access Token",
            "type": "string"
        },
        "domain": {
            "title": "Domain",
            "default": "sentry.io",
            "type": "string"
        }
    },
    "required": [
        "access_token"
    ],
    "additionalProperties": false
}
```
- PUT `/api/v1/connection/connection_key/secret`.  Use this endpoint to save the secrets to the connection config.  For example, sentry might be:
```PUT api/v1/connection/my_sentry_connector/secret```
```json
{
    "domain": "{{sentry_host}}",
    "access_token": "{{sentry_access_token}}"
}
```
- PUT `api/v1//connection/{connection_key}/test`. Use this endpoint to see if you can use the supplied secret credentials to make a test connection. In the case of saas configs, you're often making an API request to a trivial endpoint with the supplied authentication.

**Comment by eastandwestwind** [comment]
Just a note to say I've reviewed these endpoints @pattisdr and it looks good! 

Since we're going to be adding a `fides_key` (Naming TDB. We initially said `connection identifier` but something like `dataset identifier` would prob be most descriptive here) field for the datasets/saas_configs on this screen, we'll be adding additional endpoint requirements. 

We anticipate going ahead and making endpoint call(s) to save the default configs/datasets upon clicking this screen's "save" button.

The specific endpoint(s) needed to do this depends on approach for https://github.com/ethyca/fidesops/issues/814

**Comment by pattisdr** [comment]
thank you @eastandwestwind I like your suggestion of `dataset identifier`

**Comment by pattisdr** [comment]
Note: After discussion https://github.com/ethyca/fidesops/issues/814, instead of making the individual requests using the existing API endpoints to create the endpoint, and add the secrets, and later add the datasets, the "Save" button will create all of this from the fields here combined with the templates https://github.com/ethyca/fidesops/issues/814

**Comment by adriaaaa** [comment]
Ticket Breakdown:
There will be an add connection component, based on "step" it will serve as the driver to determine which component to deliver.

1. Test result display (3)
2. Left hand configuration - when user clicks on the right hand side, it will change the component + status of connector (3)
3. Everything else (name, description, secrets): this is the dynamic form (5) 
4. Tech debt ticket to throw in the backlog to handle when things break (@seanpreston to help create this one)

**Comment by seanpreston** [comment]
This is currently blocked by #814
